src/wix/WixToolset.Converters/WixConverter.cs

```diff
@@ -1080,15 +1081,17 @@ private void ConvertComponentElement(XElement element)
             var xCondition = element.Element(ConditionElementName);
             if (xCondition != null)
             {
-                if (TryGetInnerText(xCondition, out var text, out var comments) &&
+                var context = new InnerTextContext();
+
+                if (TryGetInnerText(xCondition, context) &&
```

Comment:
Upon further review, it is not clear whether you are suggesting that the new context object being returned be a copy of the context object being passed in, with changes made as necessary. This would leave the option as to whether the caller overwrites the the variable referring to the input context. Using different variables might be useful in some scenario where the original context would be reused.

You suggest later that the comment collection be initialized and exposed by the context's constructor. That seems to imply that comments recovered by previous iterations of a loop be added to a new collection each time around that loop. The current code avoids copying those comments.

This also raises the question of, "Who owns the collection?"

Reply:
I decided to take a different approach. Read on.